tests/molecular/bonds/bond/with_ids/test_with_ids.py

```diff
@@ -0,0 +1,45 @@
+from __future__ import annotations
+
+import stk
+
+from typing import Callable
+
+
+def test_with_ids(
+    bond: stk.Bond,
+    get_id_map: Callable[[stk.Bond], dict[int, int]],
+) -> None:
+    """
+    Test :meth:`.Bond.with_ids`.
+
+    Parameters:
+
+        bond : :class:`.Bond`
+            The bond to test.
+
+        get_id_map:
+            Takes a single parameter, `bond`, and returns a valid
+            `id_map` parameter for its :meth:`.Bond.with_ids`
+            method. This allows the testing of different values of
+            this parameter.
+
+    """
+
+    id_map = get_id_map(bond)
+    clone = bond.with_ids(id_map)
+    assert clone is not bond
+
+    expected_id1 = id_map.get(
+        bond.get_atom1().get_id(),
+        bond.get_atom1().get_id(),
```

Comment:
I do not see why this needs the key and value argument?

Reply:
The second argument is a defualt value to use if the id of the first atom is no found in the `id_map`. It just make it really easy to say, "if this is missing from the map, don't change the value" without having to write if statements.

Reply:
Ya I get that. I think I wrote this assuming the value would always change, but looking at the input, that is not the case, so ignore.